### PR TITLE
[SG2] Add node upgrade UI

### PR DESCRIPTION
**Pull request by dhsavell**
### Purpose of this PR

This PR ports over the node upgrade behavior from SG1.

https://user-images.githubusercontent.com/10332426/169586557-e38f4ca3-5fa4-4faa-b4fc-a9e931639547.mp4

---
### Testing status

Automated tests for upgrading nodes and warning badges on the graph:

image

TestRail cases including manual tests for prompts in the inspector:

- Upgrade prompts appear - https://qatestrail.hq.unity3d.com/index.php?/cases/view/1035868
- Upgrade prompts can be dismissed - https://qatestrail.hq.unity3d.com/index.php?/cases/view/1035869
- Node upgrading works - https://qatestrail.hq.unity3d.com/index.php?/cases/view/1035870

---
### Comments to reviewers

- Showing old versions of nodes in the searcher isn't supported yet (already tracked at [GSG-1092](https://jira.unity3d.com/browse/GSG-1092)), so the test nodes work around this with extra nested categories

### CODE REVIEW

File: com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/TestUINodes.cs [hunk 1]
```diff
@@ -524,4 +524,49 @@ internal class TestUIPropertyNode : IStandardNode
...
+            Version,
+            Name,
+            "Out = In;",
+            new ParameterDescriptor("In", TYPE.Float, GraphType.Usage.In),
+            new ParameterDescriptor("Out", TYPE.Float, GraphType.Usage.Out)
+        );
+
+        public static NodeUIDescriptor NodeUIDescriptor => new(
+            Version,
+            Name + "(V1)",
```
Comment (brett-groshong) [inline com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/TestUINodes.cs:543]: This is a new discovery! The UI descriptors match up with the function descriptors based on `Name`. They need to match. However, doing this thing makes total sense.... Should this node be named "TestUpgrade(V1)" ?

**Review by brett-groshong — APPROVED** [review]
I left 1 comment in this code but the response to it requires a later change (probably by me).

File: com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/TestUINodes.cs (hunk 1 above)
Comment (dhsavell) [reply com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/TestUINodes.cs:543]: Yeah, that was my intention-- primarily to make it apparent which version is which for testing.

Comment (brett-groshong) [reply com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/TestUINodes.cs:543]: Okay, this is an action item for me then.

**Pull request merged**